util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+require "optparse"
+
+# This script is intended to be run on a puppet infrastructure node that had the puppet_metrics_collector
+# module installed and running.  Given a time window it will collect the json files generated by the
+# puppet_metrics_collector module and tar them into the tar file.  This includes jsons that are in tar
+# files already.  Those will be extracted and included if they are in the time window.
+
+# General namespace for metrics module
+module Metrics
+  # Main class for collecting metrics files into a tar file
+  #
+  # @author Randell Pelak
+  #
+  # @attr [integer] start_epoch Epoch seconds for the start of the collection window
+  # @attr [integer] end_epoch Epoch seconds for the end of the collection window
+  # @attr [string] metrics_dir The "output directory" of the metrics collector puppet module
+  # @attr [string] output_dir The directory to place the tarfile in.
+  # @attr [string] tar_file_name Name of the tar file to create with the metrics
+  # @attr [boolean] verbose Verbose output
+  # @attr [integer] poll_interval The "collection_frequency" of the metrics collector puppet module
+  # @attr [integer] archive_interval The frequency of the *_metrics_dir cron job for the metrics
+  #   collector module.  This is how often it tars up the json files.
+  #
+  class ImportMetricsFiles
+    # Initialize class
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] results_dir The GPLT results directory to process
+    # @param [string] prefix The prefix to use when building the server tag
+    # @param [string] json2graphite_path The path to the json2graphite.rb script
+    #
+    # @return [void]
+    #
+    # @example
+    #   initialize(results_dir, prefix)
+    #
+    def initialize(results_dir, prefix, json2graphite_path)
+      @results_dir = results_dir
+      @prefix = prefix
+      @json2graphite_path = json2graphite_path
+      @id = @results_dir.split("_").last
+    end
+
+    # The main entry point to the import_metrics_files script
+    #
+    # @author Bill Claytor
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files
+      pmc_dir = "#{@results_dir}/puppet-metrics-collector"
+      raise "Directory not found: #{pmc_dir}" unless File.directory? pmc_dir
+
+      puts "Importing puppet_metrics_collector files:"
+      puts " results_dir: #{@results_dir}"
+      puts " prefix: #{@prefix}"
+      puts " id: #{@id}"
+
+      service_dirs = Dir.glob("#{pmc_dir}/*").select { |f| File.directory? f }
+
+      puts " service directories:"
+      puts service_dirs
+      puts
+
+      # each service directory
+      service_dirs.each do |service_dir|
+        puts "Checking directory: #{service_dir}"
+
+        # each host directory
+        host_dirs = Dir.glob("#{service_dir}/*").select { |f| File.directory? f }
+
+        puts " host directories: "
```

Comment:
Including the array in the string results in the array being printed rather than each element being printed on a separate line. I was not able to quickly find a way around this, so hopefully the following is sufficient:
```
puts " host directories:"; puts host_dirs; puts
```
I did have to disable the rubocop rule for this to be allowed.

Reply:
odd that the original didn't work, but could also do 
puts " host directories: \n" + host_dirs + "\n\n"
rubocop might like that better.  But doesn't matter much as long as it is only one line. :)

Reply:
just realized you were getting output were each dir got it's own line.  That's rather space heavy...  This would be better...

Reply:
Having each dir get a separate line is intentional. As above, since the json2graphite script outputs the import of each JSON file to a separate line, the only way these are even seen is if the user scrolls up through the output. If they're doing that, I think having each dir on a separate line makes this easier to read than a comma separated list that might wrap depending upon screen width.

Reply:
yeah, it's a small detail, since this should usually be one dir, except for compilers which might be a few.  As you can tell I have a thing against spammy output, as it usually means things just never get seen.  But as we already discussed, this output is so spammy already due to json2graphite no one is every going to read the output unless they are looking for something specific.  So doesn't matter as much here.